canbench-bin/src/table.rs

```diff
@@ -0,0 +1,131 @@
+use crate::data::{Change, Entry};
+
+pub(crate) fn filter_entries(data: &[Entry], noise_threshold: f64) -> Vec<Entry> {
+    let mut filtered: Vec<Entry> = data
+        .iter()
+        .filter_map(|entry| {
+            let metrics = [
+                &entry.instructions,
+                &entry.heap_increase,
+                &entry.stable_memory_increase,
+            ];
+
+            let is_significant = metrics.iter().any(|v| {
+                matches!(
+                    v.status(noise_threshold),
+                    Change::New | Change::Improved | Change::Regressed
+                )
+            });
+
+            if !is_significant {
+                return None;
+            }
+
+            let mut status = String::new();
+            if entry.status.is_empty() {
+                if metrics
+                    .iter()
+                    .any(|v| v.status(noise_threshold) == Change::Regressed)
+                {
+                    status.push('+');
+                }
+                if metrics
+                    .iter()
+                    .any(|v| v.status(noise_threshold) == Change::Improved)
+                {
+                    if !status.is_empty() {
+                        status.push('/');
+                    }
+                    status.push('-');
+                }
+            } else {
+                status = entry.status.clone();
+            }
+
+            let mut updated = entry.clone();
+            updated.status = status;
+            Some(updated)
+        })
+        .collect();
+
+    // Sort by name, ascending.
+    filtered.sort_by(|a, b| a.benchmark.full_name().cmp(&b.benchmark.full_name()));
+    // Sort by status.
+    filtered.sort_by(|a, b| a.status.cmp(&b.status));
+    // Sort by instructions percent diff, descending.
+    const EMPTY: f64 = f64::MIN;
+    filtered.sort_by(|a, b| {
+        a.instructions
+            .percent_diff()
+            .unwrap_or(EMPTY)
+            .partial_cmp(&b.instructions.percent_diff().unwrap_or(EMPTY))
+            .unwrap_or(std::cmp::Ordering::Equal)
+            .reverse()
+    });
+
+    filtered
+}
+
+pub(crate) fn print_table(data: &[Entry]) {
+    let columns = [
+        "status", "name", "ins", "ins Δ%", "HI", "HI Δ%", "SMI", "SMI Δ%",
```

Comment:
How about HeapInc and StableMemInc?

Reply:
I am trying to fit into github comment width, which may not look too bad for canbench repo, but might be in stable-structures, because of long benchmark names. So I'd prefer to keep it as is, especially because short summary does have full names. And also CSV file has full names as well.

![image](https://github.com/user-attachments/assets/8c4ec4ba-ba85-482b-8002-a884300596b6)